### PR TITLE
chore: update debian base to buster-v1.8.0 

### PR DESCRIPTION
**Pull request by aramase**
**What this PR does / why we need it**:
- Updates debian base to `buster-v1.8.0` that contains fix for vulnerabilities
- Adds `IMAGE_VERSION` to build-arg to pass the version to go build 

**Which issue(s) this PR fixes** *(optional, using `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when the PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: docker/windows.Dockerfile [hunk 1]
```diff
@@ -8,6 +8,7 @@ WORKDIR /go/src/sigs.k8s.io/secrets-store-csi-driver
 ADD . .
 ARG TARGETARCH
 ARG TARGETOS
+ARG IMAGE_VERSION
```
Comment (nilekhc) [inline docker/windows.Dockerfile:11]: Do we use this ARG inside dockerfile?

Comment (aramase) [reply docker/windows.Dockerfile:11]: The ARG is passed to the builder and will get consumed as part of `make build`. If not set, the `IMAGE_VERSION` provided as build will not take effect instead the `IMAGE_VERSION` will Makefile will be used.

**Review by nilekhc — APPROVED** [review]

**Pull request merged**